### ISSUE TITLE
Support wildcard DNS records

### ISSUE THREAD

**Issue by neigor** [skynetservices/skydns#105]
I could not figure out how to add wildcard DNS records (http://en.wikipedia.org/wiki/Wildcard_DNS_record).

It is unfortunate to another DNS server just for this ;(

**Comment by AlessandroEmm** [comment]
Hi All

I'm not sure whether my question goes along with @neigor 's but why are Entries in the DNS overwritten when two services on two separate machines register the same name? I would have expected it to build a mappings name -> List of IPs.

Thanks
Alessandro

**Comment by miekg** [comment]
(aside: i actually need to look in the pull request again).

Anyhow. Yes entries for a single key are overwritten, this is how it works. If a name has a new ip you write the new ip, there not appended.

Basically: you're doing it wrong :) If you want to machine register the same name, then machine a needs to registar a.bla.skydns.local and machine b needs to registar b.bla.skydns.local. You can then use bla.skydns.local is the *shared* name.

**Comment by MartinSoto** [comment]
Hi!

I guess I have another use case for this. We have a Web application that hosts domains for our customers, e.g.:

Main domain: someservice.com
Customer 1: ibm.someservice.com
Customer 2: bmw.someservice.com
...

New customers are simply entered into the database and their subdomain name is just stored together with the rest of their data. Accessing a customer's domain will make the application behave differently (e.g., display a different logo and colors) but it's still the same application, not a separate instance of it or anything. The behavioral differences are all in the business logic, so we definitely don't want to start a separate service in order to handle each customer.

We built a Docker-based test environment for running this application (together with a bunch of related services) around skydns, and it's working very well. Now we wanted to add support for customized customer domains, but this limitation is making it quite hard to do so. Of course, we can still add a cron job that pulls the subdomains from the database and adds entries to skydns, but a simple wildcard domain will be easier and a lot more reliable.

**Comment by miekg** [comment]
Yes, a feature like this needs to go in. The only thing I want to make sure is that we won't make an extra query when we get back 'not found'.

**Comment by Johnxiaoyi** [comment]
Do we have any progress on the wildcard support?

**Comment by miekg** [comment]
No progress, because the double query issues hasn't been fleshed out (yet).

**Comment by Johnxiaoyi** [comment]
@miekg thanks.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [skydns] Support wildcard DNS r..." ]
>@miekg thanks.

np, Also nobody is *actually* fleshing this out; haven't seen any new proposals.

/Miek

--
Miek Gieben
